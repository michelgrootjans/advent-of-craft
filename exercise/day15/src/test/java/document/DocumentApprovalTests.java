package document;

import static org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations;

import org.approvaltests.reporters.ClipboardReporter;
import org.approvaltests.reporters.Junit5Reporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

@UseReporter({ClipboardReporter.class, Junit5Reporter.class})
class DocumentApprovalTests {
    @Test
    void approval_test_of_document_and_record_type() {
        verifyAllCombinations(
            DocumentTemplateType::fromDocumentTypeAndRecordType,
            new String[]{"DEER", "AUTP", "AUTM", "SPEC", "GLPP", "GLPM", "INVALID"},
            new String[]{"INDIVIDUAL_PROSPECT", "LEGAL_PROSPECT", "ALL", "INVALID"}
        );
    }
}
