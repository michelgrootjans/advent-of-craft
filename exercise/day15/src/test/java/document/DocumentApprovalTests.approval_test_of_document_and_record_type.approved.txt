[DEER, INDIVIDUAL_PROSPECT] => DEERPP 
[DEER, LEGAL_PROSPECT] => DEERPM 
[DEER, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[DEER, INVALID] => java.lang.IllegalArgumentException: No enum constant document.RecordType.INVALID 
[AUTP, INDIVIDUAL_PROSPECT] => AUTP 
[AUTP, LEGAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[AUTP, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[AUTP, INVALID] => java.lang.IllegalArgumentException: No enum constant document.RecordType.INVALID 
[AUTM, INDIVIDUAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[AUTM, LEGAL_PROSPECT] => AUTM 
[AUTM, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[AUTM, INVALID] => java.lang.IllegalArgumentException: No enum constant document.RecordType.INVALID 
[SPEC, INDIVIDUAL_PROSPECT] => SPEC 
[SPEC, LEGAL_PROSPECT] => SPEC 
[SPEC, ALL] => SPEC 
[SPEC, INVALID] => java.lang.IllegalArgumentException: No enum constant document.RecordType.INVALID 
[GLPP, INDIVIDUAL_PROSPECT] => GLPP 
[GLPP, LEGAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[GLPP, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[GLPP, INVALID] => java.lang.IllegalArgumentException: No enum constant document.RecordType.INVALID 
[GLPM, INDIVIDUAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[GLPM, LEGAL_PROSPECT] => GLPM 
[GLPM, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[GLPM, INVALID] => java.lang.IllegalArgumentException: No enum constant document.RecordType.INVALID 
[INVALID, INDIVIDUAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[INVALID, LEGAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[INVALID, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[INVALID, INVALID] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
